src/main/groovy/io/seqera/wave/service/builder/BuildStrategy.groovy

```diff
@@ -93,7 +93,7 @@ abstract class BuildStrategy {
         result
             << 'sh'
             << '-c'
-            << "singularity build image.sif ${req.workDir}/Containerfile && singularity push image.sif ${req.targetImage}".toString()
+            << "singularity build image.sif ${req.workDir}/Containerfile --platform ${req.platform.toString()} && singularity push image.sif ${req.targetImage}".toString()
```

Comment:
I cannot find `--platform` option in the singularity build command. What am I missing 

```
docker run quay.io/singularity/singularity:v4.0.1 build -h
```

Reply:
I found this flag in docs https://github.com/sylabs/singularity/releases/tag/v4.0.0

Reply:
with --platform
```
14:23:35.344 [wave-builder-1] DEBUG i.s.w.s.builder.DockerBuildStrategy - Build run command: docker run --rm --privileged --entrypoint  -v /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261:/Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261 -v /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261/config.json:/root/.singularity/docker-config.json:ro -v /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261/singularity-remote.yaml:/root/.singularity/remote.yaml:ro --platform linux/arm64 quay.io/singularity/singularity:v4.0.1-slim-arm64 sh -c singularity build image.sif /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261/Containerfile --platform linux/arm64 && singularity push image.sif oras://docker.io/hrma017/dev:cowsay--93569c0c19462261
14:23:35.654 [wave-builder-1] INFO  i.s.w.s.b.ContainerBuildServiceImpl - == Build request 93569c0c19462261 completed with status=1
```

with --arch
```
14:19:34.175 [wave-builder-1] DEBUG i.s.w.s.builder.DockerBuildStrategy - Build run command: docker run --rm --privileged --entrypoint  -v /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261:/Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261 -v /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261/config.json:/root/.singularity/docker-config.json:ro -v /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261/singularity-remote.yaml:/root/.singularity/remote.yaml:ro --platform linux/arm64 quay.io/singularity/singularity:v4.0.1-slim-arm64 sh -c singularity build image.sif /Users/munish.chouhan/main_ground/wave/284/wave/build-workspace/93569c0c19462261/Containerfile --arch arm64 && singularity push image.sif oras://docker.io/hrma017/dev:cowsay--93569c0c19462261
14:19:34.467 [wave-builder-1] INFO  i.s.w.s.b.ContainerBuildServiceImpl - == Build request 93569c0c19462261 completed with status=1
```